### PR TITLE
Fix bug in import in LinearAlgebra::distributed::Vector

### PR DESCRIPTION
**Pull request by Rombur**
I tried to be smart and minimize the amount of data which is send when using `import` and of course it doesn't work all the time. The new implementation works even when the `ReadWriteVector` and the locally owned elements of the `LA::d::Vector` do not overlap at all.

### CODE REVIEW

File: include/deal.II/lac/la_parallel_vector.templates.h [hunk 1]
```diff
@@ -683,28 +727,63 @@ namespace LinearAlgebra
                       ExcMessage("The communication pattern is not of type "
                                  "Utilities::MPI::Partitioner."));
         }
+      IndexSet ghost_indices(V.get_stored_elements());
+      ghost_indices.subtract_set(locally_owned_elem);
+      IndexSet local_indices(locally_owned_elem);
+      local_indices.print(std::cout);
+      ghost_indices.print(std::cout);
```
Comment (kronbichler) [inline include/deal.II/lac/la_parallel_vector.templates.h]: Can you please remove this debug output?

File: include/deal.II/lac/la_parallel_vector.templates.h [hunk 2]
```diff
@@ -669,8 +712,9 @@ namespace LinearAlgebra
           // then create the communication pattern
           IndexSet ghost_indices(V.get_stored_elements());
           ghost_indices.subtract_set(locally_owned_elem);
-          IndexSet local_indices(V.get_stored_elements());
-          local_indices.subtract_set(ghost_indices);
+          IndexSet local_indices(locally_owned_elem);
+          // IndexSet local_indices(V.get_stored_elements());
+          // local_indices.subtract_set(ghost_indices);
```
Comment (kronbichler) [inline include/deal.II/lac/la_parallel_vector.templates.h]: Can you please remove the commented-out code?

**Comment by Rombur** [comment]
Oops, sorry about that. It's fixed now.

**Pull request merged**